x-pack/plugin/downsample/src/main/java/org/elasticsearch/xpack/downsample/DownsampleShardPersistentTaskExecutor.java

```diff
@@ -159,6 +164,29 @@ public PersistentTasksCustomMetadata.Assignment getAssignment(
             .orElse(NO_NODE_FOUND);
     }
 
+    /**
+     * An eligible node to run the downsampling task for a shard is a node that holds
+     * a searchable version of this shard.
+     * In stateless deployment we choose only nodes that hold search shards.
+     * Otherwise, we choose the node that holds the primary shard.
+     * Visible for testing.
+     * @param indexShardRouting the routing of the shard to be downsampled
+     * @return the set of candidate nodes downsampling can run on.
+     */
+    Set<String> getEligibleNodes(IndexShardRoutingTable indexShardRouting) {
+        if (isStateless) {
+            return indexShardRouting.assignedShards()
+                .stream()
+                .filter(shardRouting -> shardRouting.primary() == false && shardRouting.started())
```

Comment:
maybe we can use `isPromotableToPrimary()` instead, just in case that we have multiple indexing nodes per shard in the future (i.e. standby indexing nodes).